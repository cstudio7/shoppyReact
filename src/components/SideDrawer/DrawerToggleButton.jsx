import React from 'react';
import PropTypes from 'prop-types';
import './styles/DrawerToggleButton.scss';

const drawerToggleButton = props => (
  <button type="button" className="toggle-button" onClick={props.click}>
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
  </button>
);

drawerToggleButton.propTypes = {
  click: PropTypes.func.isRequired
};

export default drawerToggleButton;
