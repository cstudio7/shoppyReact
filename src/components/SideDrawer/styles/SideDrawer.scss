.side-drawer {
  height: 100%;
  background: white;
  position: fixed;
  top: 0;
  left: 0;
  width: 70%;
  max-width: 400px;
  z-index: 200;
  transform: translateX(-100%);
  transition: transform 0.3s ease-out;
}

.side-drawer.open {
  transform: translateX(0);
}

.side-drawer ul {
  height: 100%;
  list-style: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.side-drawer li {
  margin: 0.9rem 0;
}

.side-drawer a {
  color: #ff6300;
  text-decoration: none;
  font-size: 1.2rem;
}

.side-drawer a:hover,
.side-drawer a:active {
  display: inline-block;
  text-decoration: none;
  padding: 13px 23px;
  font-size: 18px;
  -webkit-transition: 0.2s all;
  transition: 0.5s all;
  &:hover {
    color: #fff;
    background: #ff6300;
  }
  &:active {
    color: #fff;
    background: #ff6300;
  }
}

@media (min-width: 769px) {
  .side-drawer {
    display: none;
  }
}
