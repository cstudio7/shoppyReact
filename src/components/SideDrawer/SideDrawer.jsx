import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NavLinks from '../NavLinks';
import './styles/SideDrawer.scss';

class SideDrawer extends Component {
  render() {
    let drawerClasses = 'side-drawer';
    if (this.props.show) {
      drawerClasses = 'side-drawer open';
    }
    return (
      <nav className={drawerClasses}>
        <NavLinks userRole={this.props.userRole} />
      </nav>
    );
  }
}

SideDrawer.propTypes = {
  show: PropTypes.bool,
  userRole: PropTypes.string.isRequired
};

SideDrawer.defaultProps = {
  show: false
};

export default SideDrawer;
