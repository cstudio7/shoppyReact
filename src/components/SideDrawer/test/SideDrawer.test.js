import React from 'react';
import { shallow } from 'enzyme';
import SideDrawer from '../SideDrawer';

describe('Test the SideDrawer Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SideDrawer show />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<SideDrawer show={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
