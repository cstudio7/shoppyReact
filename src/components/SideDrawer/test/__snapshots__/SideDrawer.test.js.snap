// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the SideDrawer Component should match snapshot 1`] = `
<nav
  className="side-drawer open"
>
  <AdminNav />
</nav>
`;

exports[`Test the SideDrawer Component should match snapshot 2`] = `
<nav
  className="side-drawer"
>
  <AdminNav />
</nav>
`;
